transportable-udfs-trino-plugin/src/main/java/com/linkedin/transport/trino/TransportUDFClassLoader.java

```diff
@@ -0,0 +1,45 @@
+/**
+ * Copyright 2019 LinkedIn Corporation. All rights reserved.
+ * Licensed under the BSD-2 Clause license.
+ * See LICENSE in the project root for license information.
+ */
+package com.linkedin.transport.trino;
+
+import java.net.URL;
+import java.net.URLClassLoader;
+import java.util.List;
+
+
+public class TransportUDFClassLoader extends URLClassLoader {
+  private final ClassLoader parent;
+
+  public TransportUDFClassLoader(ClassLoader parent, List<URL> urls) {
+    super(urls.toArray(new URL[0]));
+    this.parent = parent;
+  }
+
+  @Override
+  public Class<?> loadClass(String name, boolean resolve) throws ClassNotFoundException {
+    synchronized (getClassLoadingLock(name)) {
+      // Check if class is in the loaded classes cache
+      Class<?> cachedClass = findLoadedClass(name);
+      if (cachedClass != null) {
+        return resolveClass(cachedClass, resolve);
+      }
+
+      if (name.endsWith("StdUdfWrapper") || name.startsWith("com.linkedin.transport.api")) {
```

Comment:
what classes are we trying to catch with the `endsWith("StdUdfWrapper")`? examples?

Reply:
I think matching the full class name here would be better. `com.linkedin.transport.trino.StdUdfWrapper`

https://github.com/linkedin/transport/blob/master/transportable-udfs-trino/src/main/java/com/linkedin/transport/trino/StdUdfWrapper.java

Reply:
Yes, the full class path of the class are added to make it clear.